### PR TITLE
feat(BDSGOLD-771): Applying SPARK-24570 and SPARK-24196 patch and replacing Spark hive code from BDS hive

### PR DESCRIPTION
**Pull request by alee-altiscale**
## What changes were proposed in this pull request?

- Applying SPARK-24570 and SPARK-24196 patch to enable metadata browsing via Thrift channel (e.g. JDBC-ODBC bridge Type 1 JDBC driver).
- Backport BDS hive-1.2.1 changes to sync with spark hive-thriftserver hive-service code base (this also includes hadoop-auth for token authentication support added from previous changes)

## How was this patch tested?

Applying BDS test suite and FoA with manual test.

### CODE REVIEW

File: sap-xmake-build.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ export BUILD_TIME=$(date +%Y%m%d%H%M)
 export MAVEN_OPTS=${MAVEN_OPTS:-"-Xmx2048m -XX:MaxPermSize=1024m"}
 export PRODUCTION_RELEASE=${PRODUCTION_RELEASE:-"false"}
 
-export PACKAGE_BRANCH=${PACKAGE_BRANCH:-"branch-2.3.2-alti"}
+export PACKAGE_BRANCH=${PACKAGE_BRANCH:-"sap-branch-2.3.2-alti-SPARK-24570-patch"}
```
Comment (alee-altiscale) [inline sap-xmake-build.sh:49]: I will revert this back to `branch-2.3.2-alti` before merging.

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/TableTypeMapping.java [hunk 2]
```diff
@@ -27,18 +27,18 @@
...
    * @return
    */
-  String mapToHiveType(String clientTypeName);
+  public String mapToHiveType (String clientTypeName);
 
   /**
    * Map hive's table type name to client's table type
-   * @param hiveTypeName
+   * @param clientTypeName
    * @return
```
Comment (sarah-altiscale) [inline sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/TableTypeMapping.java:35]: I guess this is pulled in but hiveTypeName is the right param in the doc not clientTypeName

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/SQLOperation.java [hunk 3]
```diff
@@ -443,7 +456,7 @@ private SerDe getSerDe() throws SQLException {
   private HiveConf getConfigForOperation() throws HiveSQLException {
     HiveConf sqlOperationConf = getParentSession().getHiveConf();
     if (!getConfOverlay().isEmpty() || shouldRunAsync()) {
-      // clone the parent session config for this query
+      // clone the partent session config for this query
```
Comment (sarah-altiscale) [inline sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/SQLOperation.java:459]: Change has the sp LOL. But this is pulled in so we probably cannot help it

Comment (alee-altiscale) [reply sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/SQLOperation.java:459]: :( I know... either we fix it in hive 1.2.1 repo or here, but here, future changes will surface again :(

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/TableTypeMapping.java (hunk 2 above)
Comment (alee-altiscale) [reply sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/TableTypeMapping.java:35]: :( I know... either we fix it in hive 1.2.1 repo or here, but here, future changes will surface again :(. spark 2.3.2 imported hive 1.2.x+ and it is newer than our hive 1.2.1. :(

**Review by sarah-altiscale — APPROVED** [review]

**Pull request merged**